### PR TITLE
Creating new feats

### PR DESCRIPTION
**Pull request by adrissonGalvao**

### CODE REVIEW

File: repository/user.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	if err != nil {
+		log.Fatal(err)
+	}
+	u.session = session
+}
+func (u *UserRepository) InsertUser(user domain.User) error {
+	sess := u.session.Copy()
+	defer sess.Close()
+
+	err := sess.DB(u.database).C("user").Insert(&user)
```
Comment (edvmorango) [inline repository/user.go:34]: @adrissonGalvao this "err" receives the result of "Insert"? This method gives feedback when "Insert" fails?

Comment (adrissonGalvao) [reply repository/user.go:34]: @edvmorango if an operation returns error so it send error else it returns nil. Yes it sends error feedback

File: service/user.go [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	defer r.Body.Close()
+	var user domain.User
+
+	if err := json.NewDecoder(r.Body).Decode(&user); err != nil {
+		respondWithError(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+
+	user.ID = bson.NewObjectId()
+	if err := userRepository.InsertUser(user); err != nil {
```
Comment (adrissonGalvao) [inline service/user.go:28]: @edvmorango if  not insert so operation fail and return error else all right

Comment (edvmorango) [reply service/user.go:28]: Sure @adrissonGalvao .

**Pull request merged**